src/main/java/datawave/query/util/MetadataHelper.java

```diff
@@ -1215,6 +1255,38 @@ public Long getCountsByFieldInDayWithTypes(String fieldName, String date, final
     protected HashMap<String,Long> getCountsByFieldInDayWithTypes(Entry<String,String> identifier) throws TableNotFoundException, IOException {
         String fieldName = identifier.getKey();
         String date = identifier.getValue();
+        boolean foundCompressedValue = false;
+        
+        Scanner scanner = ScannerHelper.createScanner(connector, metadataTableName, auths);
+        scanner.fetchColumnFamily(ColumnFamilyConstants.COLF_F);
+        scanner.setRange(Range.exact(fieldName));
+        
+        IteratorSetting cqRegex = new IteratorSetting(50, RegExFilter.class);
+        RegExFilter.setRegexs(cqRegex, null, null, MetadataHelper.COL_QUAL_PREFIX + ".*", null, false);
+        scanner.addScanIterator(cqRegex);
+        
+        final HashMap<String,Long> datatypeToCounts = Maps.newHashMap();
+        for (Entry<Key,Value> countEntry : scanner) {
+            foundCompressedValue = true;
+            DateFrequencyValue dateFrequencyValue = new DateFrequencyValue();
+            TreeMap<YearMonthDay,Frequency> dateFrequencies = dateFrequencyValue.deserialize(countEntry.getValue());
```

Comment:
Will this handle non-compressed / aggregated entries?

Reply:
You are correct.  It won't find the legacy style entries. That is why I call the old function in these lines below:
` if (!foundCompressedValue)
            datatypeToCounts.putAll(getCountsByFieldInDayWithTypesLegacy(identifier));`

It was the easiest way I saw to make a foolproof support for legacy data that was clean and easy for me to implement.

Reply:
Lets remove the legacy stuff here.  If we are going to assume everything is aggregated, then lets assume it and got forward.